xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/TemplateProviderSheet.xml

```diff
@@ -42,56 +42,52 @@
   <content>{{velocity}}
 #set ($discard = $xwiki.jsx.use('XWiki.TemplateProviderSheet'))
 #set ($providerObj = $doc.getObject('XWiki.TemplateProviderClass'))
-#if ($editor == 'inline')
-  ; {{translation key="xe.templateprovider.name"/}}
-  : {{html}}&lt;input type="text" value="$!{doc.getTitle()}" name="title" /&gt;{{/html}}
-#end
-; {{translation key="xe.templateprovider.templatename"/}}
-#if ($editor == 'inline')
+{{html wiki="true"}}
```

Comment:
I agree with both of you :)

Note that having HTML macro means that some Renderers in the future may not work correctly (since HTML Macro generates Raw Blocks). That's unless we decide that all Renderers have to support Raw Blocks. This is a bit theoretical at this point since our main Renderer is the (X)HTML Renderer and it supports HTML obviously ;) If we were to write a PDF Renderer based on iText for example (I started one in a branch in the Rendering repo a long time ago), the HTML would need some special handling to convert it to HTML (as we currently do our current PDF export,- which doesn't go through XWiki Rendering).

Just to say that till now our strategy was to reduce to the maximum the usage of the HTML Macro and even to be able to not use it at all by introducing some new Macros, such as a Form Macro for example for designing forms (PDF has Form support too for example). Note that this would probably need Rendering v3, to allow adding new Events.

Now, I agree that HTML is better for UI than wiki syntax is so I think we need to discuss this on the list and set up a strategy as to when it's ok to use the HTML Macro and when it's not and the consequences of that. One idea could be that it's ok that pages with UI may not have proper Rendering for some Renderers (ie ignore the Raw Blocks).